module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -780,243 +804,110 @@ namespace app
-        if (getCurrentWindow()->getName() == popupName) {
-            returnToPreviousWindow();
-        }
-        else {
-            popWindow(popupName);
-        }
+        LOG_INFO("abort popup: %s from window %s", popupName.c_str(), getCurrentWindow()->getName().c_str());
+        windowsStack().popTo(popupName);
+        returnToPreviousWindow();
     }
 
-    bool ApplicationCommon::isPopupPermitted([[maybe_unused]] gui::popup::ID popupId) const
+    bool ApplicationCommon::handleUI_DBNotification(sys::Message *msg, const UiNotificationFilter &filter)
     {
-        return true;
-    }
-
-    bool ApplicationCommon::popToWindow(const std::string &window)
-    {
-        if (window == gui::name::window::no_window) {
-            bool ret = false;
-            if (windowsStack.stack.size() <= 1) {
-                windowsStack.stack.clear();
-                ret = true;
+        bool handled = false;
+        for (const auto &[name, window] : windowsStack()) {
+            if (filter == nullptr || (filter != nullptr && filter(msg, window->getName()))) {
+                handled |= window->onDatabaseMessage(msg);
             }
-            return ret;
         }
-
-        auto ret = std::find(windowsStack.stack.begin(), windowsStack.stack.end(), window);
-        if (ret != windowsStack.stack.end()) {
-            LOG_INFO("Pop last window(s) [%d] :  %s",
-                     static_cast<int>(std::distance(ret, windowsStack.stack.end())),
-                     ret->c_str());
-            windowsStack.stack.erase(std::next(ret), windowsStack.stack.end());
-            LOG_INFO("Curent window... %s vs %s", ret->c_str(), windowsStack.stack.back().c_str());
-            return true;
-        }
-        return false;
+        return handled;
     }
 
-    void ApplicationCommon::pushWindow(const std::string &newWindow)
+    void ApplicationCommon::pushWindow(const std::string &newWindow, const gui::popup::Disposition &d)
     {
         // handle if window was already on
-        LOG_DEBUG("App: %s window %s request", GetName().c_str(), newWindow.c_str());
-        if (popToWindow(newWindow)) {
+        LOG_INFO("App: %s window %s request", GetName().c_str(), newWindow.c_str());
+        if (newWindow.empty() && windowsStack().popLastWindow()) {
+            LOG_INFO("Empty window request!");
             return;
         }
-        else {
-            windowsStack.push(newWindow, windowsFactory.build(this, newWindow));
-        }
-#if DEBUG_APPLICATION_MANAGEMENT == 1
-        LOG_DEBUG("[%d] newWindow: %s", (int)windowsStack.stack.size(), newWindow.c_str());
-        for (auto &el : windowsStack.stack) {
-            LOG_DEBUG("-> %s", el.c_str());
+        if (windowsStack().popTo(newWindow)) {
+            LOG_INFO("Get back to window!");
+            return;
         }
-#endif
-    };
+        LOG_INFO("Create window for stack: %s", newWindow.c_str());
+        windowsStack().push(newWindow, windowsFactory.build(this, newWindow), d);
```

Comment:
Probably these logs should be `DEBUG`

Reply:
if you look closely all the logs for that part are in INFO, not in DEBUG :) I can agree that it should be handled somehow, but I suppose that we should make a decision and change all logs at once for this part of code.